core-plugins/src/test/java/co/cask/hydrator/plugin/batch/joiner/JoinFieldLineageTest.java

```diff
@@ -0,0 +1,165 @@
+                                             Arrays.asList("id", "customer_id")));
+    expected.add(new FieldTransformOperation("Identity customer.name", Joiner.IDENTITY_OPERATION_DESCRIPTION,
+                                             Collections.singletonList("customer.name"),
+                                             Collections.singletonList("name")));
+    expected.add(new FieldTransformOperation("Identity purchase.item", Joiner.IDENTITY_OPERATION_DESCRIPTION,
+                                             Collections.singletonList("purchase.item"),
+                                             Collections.singletonList("item")));
+    Assert.assertEquals(expected, fieldOperations);
+  }
+
+  @Test
+  public void testSimpleJoinWithRenameJoinKeys() {
+    //  customer -> (id, name)----------
+    //                                  |
+    //                                JOIN  ------->(id_from_customer, id_from_purchase, name, item)
+    //                                  |
+    //  purchase ->(customer_id, item)---
+
+    List<Joiner.OutputFieldInfo> outputFieldInfos = new ArrayList<>();
+    outputFieldInfos.add(new Joiner.OutputFieldInfo("id_from_customer", "customer", "id",
+                                                    Schema.Field.of("id", Schema.of(Schema.Type.INT))));
+    outputFieldInfos.add(new Joiner.OutputFieldInfo("id_from_purchase", "purchase", "customer_id",
+                                                    Schema.Field.of("customer_id", Schema.of(Schema.Type.INT))));
+    Map<String, List<String>> perStageJoinKeys = new LinkedHashMap<>();
+    perStageJoinKeys.put("customer", Collections.singletonList("id"));
+    perStageJoinKeys.put("purchase", Collections.singletonList("customer_id"));
+
+    List<FieldOperation> fieldOperations = Joiner.createFieldOperations(outputFieldInfos, perStageJoinKeys);
+
+    List<FieldOperation> expected = new ArrayList<>();
+    expected.add(new FieldTransformOperation("Join", "Join Operation",
+                                             Arrays.asList("customer.id", "purchase.customer_id"),
+                                             Arrays.asList("id", "customer_id")));
+    expected.add(new FieldTransformOperation("Rename id", Joiner.RENAME_OPERATION_DESCRIPTION,
+                                             Collections.singletonList("id"),
+                                             Collections.singletonList("id_from_customer")));
+    expected.add(new FieldTransformOperation("Rename customer_id", Joiner.RENAME_OPERATION_DESCRIPTION,
+                                             Collections.singletonList("customer_id"),
+                                             Collections.singletonList("id_from_purchase")));
+    Assert.assertEquals(expected, fieldOperations);
+  }
+
+  @Test
+  public void testSimpleJoinWithRenameOnAdditionalFields() {
+    //  customer -> (id, name)----------
```

Comment:
can you also add a test case where 3 inputs are being joined

Reply:
added.